src/main/java/httpServer/ListenerAndResponder.java

```diff
@@ -40,46 +43,114 @@ public Socket keepListening() throws IOException {
     }
 
     public void readClientInput(Socket clientSocket) throws IOException {
-
+        System.err.println("Starting to read");
         ClientReadable bufferedReader = clientReaderFactory.makeReader(clientSocket);
-
-        String httpRequest;
-        httpRequest = readAllLines(bufferedReader);
+        HTTPRequest httpRequest;
+        httpRequest = readAllLinesAndReturnHttpRequest(bufferedReader);
         if (httpRequest == null) {
             StatusCodes statusCodes = StatusCodes.BAD_REQUEST;
             String responseText = statusCodes.httpResponse;
             printServerResponse(responseText, clientSocket);
         }
         if (!httpRequest.equals("")) {
-            HTTPRequest request = RequestParser.parse(httpRequest);
-            System.out.println(httpRequest);
-            String httpResponse = router.getResponse(request);
-            printServerResponse(httpResponse, clientSocket);
+            HTTPResponse httpResponse = router.getResponse(httpRequest);
+            String responseString = httpResponse.toString();
+            System.err.println(responseString.length());
+            printServerResponse(responseString, clientSocket);
         }
 
     }
 
-    private void printServerResponse(String message, Socket clientSocket) throws IOException {
-            ClientWriteable printer = clientWriterFactory.makePrinter(clientSocket);
-            System.out.println(message);
-            printer.println(message);
-
+    private void printServerResponse(String httpResponse, Socket clientSocket) throws IOException {
+        ClientWriteable printer = clientWriterFactory.makePrinter(clientSocket);
+        System.out.println(httpResponse);
+        printer.print(httpResponse);
+        printer.close();
     }
 
-    private String readAllLines(ClientReadable bufferedReader)  {
-        String line;
-        StringBuilder stringBuilder = new StringBuilder();
+    private HTTPRequest readAllLinesAndReturnHttpRequest(ClientReadable bufferedReader)  {
```

Comment:
This method is very long, I'd like you to break it up into more readable fragments.

That "And" in the method name is kind of a smell that the method has more than a single responsibility.